tests/test_helpers/test_dataset_utils.py

```diff
@@ -0,0 +1,667 @@
+from tests.test_helpers.dataset_utils import generate_collections, generate_dataset
+
+
```

Comment:
nice job with these tests - they're a great starting point for understanding the expected functionality of the utilities.

one follow up - would it be feasible/useful to include more of an "integration" level test that actually tests and demonstrates how the utility can work on a _real_ API response, i.e. using a saas provider we already integrate with as an example? that test would obviously need to be marked as an `integration_saas` test since it would require external communication, but just seems like it could be a nice-to-have. (if you think this is feasible, maybe we can split out a separate issue/PR for it, since i don't think it should necessarily block this PR)